packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -721,75 +750,34 @@ export class NxMonorepoProject extends TypeScriptProject {
   }
 
   /**
-   * Updates the install task for python projects so that they are run serially and in dependency order such that python
-   * projects within the monorepo can declare dependencies on one another.
+   * Ensures the install task for non-node projects is executed postinstall.
    * @private
    */
-  private wirePythonDependencies() {
-    // Find any python projects
-    const pythonProjects = this.subProjects.filter(
-      (project) => project instanceof PythonProject
-    ) as PythonProject[];
-
-    if (pythonProjects.length === 0) {
-      // Nothing to do for no python projects
-      return;
-    }
-
-    // Move all install commands to install-py so that they are not installed in parallel by the monorepo package manager.
-    // eg yarn install will run the install task for all packages in parallel which can lead to conflicts for python.
-    pythonProjects.forEach((pythonProject) => {
-      const installPyTask =
-        pythonProject.tasks.tryFind("install-py") ??
-        pythonProject.addTask("install-py");
-      const installTask = pythonProject.tasks.tryFind("install");
-
-      (installTask?.steps || []).forEach((step) => {
-        this.copyStepIntoTask(step, installPyTask, pythonProject);
-      });
-
-      installTask?.reset();
-    });
+  private installNonNodeDependencies() {
+    const installProjects = this.subProjects.filter(
+      (project) =>
+        !(project instanceof NodeProject) && project.tasks.tryFind("install")
+    );
 
-    // Add an install task to the monorepo to include running the install-py command serially to avoid conflicting writes
-    // to a shared virtual env. This is also managed by nx so that installs occur in dependency order.
     const monorepoInstallTask =
-      this.tasks.tryFind("install") ?? this.addTask("install");
+      this.tasks.tryFind("postinstall") ?? this.addTask("postinstall");
     monorepoInstallTask.exec(
-      `${buildExecutableCommand(
-        this.package.packageManager
-      )} nx run-many --target install-py --projects ${pythonProjects
-        .map((project) => project.name)
-        .join(",")} --parallel=1`
+      buildExecutableCommand(
+        this.package.packageManager,
+        `nx run-many --target install --projects ${installProjects
```

Comment:
Much cleaner, nice one!

We should remove this too: https://github.com/aws/aws-prototyping-sdk/blob/mainline/packages/open-api-gateway/src/project/codegen/generated-python-client-project.ts#L65-L75

It's there because open-api-gateway synths stuff early, before the monorepo can make changes to its tasks... No such hack in type-safe-api though haha

Reply:
Updated. Good call